mlflow/sklearn/__init__.py

```diff
@@ -948,6 +1041,52 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
+            if hasattr(class_def, func_name):
+                safe_patch(
+                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                )
+
+    from sklearn import metrics
+    # Note:
+    #  don't need patch model.score() because internally model.score() will call the concrete metric API
+    for metric_method_name in metrics.__all__:
+        # excludes '**Display' object such as `ConfusionMatrixDisplay`
+        if callable(getattr(metrics, metric_method_name)) and not metric_method_name.endswith('Display'):
+            safe_patch(FLAVOR_NAME, metrics, metric_method_name, patched_metric_api, manage_run=False)
+
+    def patched_scorer_call(original, self, *args, **kwargs):
+        global _autolog_training_status
+
+        try:
+            _autolog_training_status.in_scorer_call_scope = True
+            metric = original(self, *args, **kwargs)
+        finally:
+            _autolog_training_status.in_scorer_call_scope = False
+
+        if not _autolog_training_status.in_fit_call_scope and \
+                not _autolog_training_status.in_eval_and_log_metrics_scope:
+            estimator = args[0]
+            # TODO: refine metric_name to make it include arguments set for the metric
```

Comment:
Does this mean when a user creates the followng scorer,

```
scorer = make_scorer(fbeta_score, beta=2)
```

include `beta=2` in the metric_name?

Reply:
yes.

Reply:
One concern is this makes a metric name long and some users may not like that.